server/src/main/java/org/apache/druid/segment/realtime/firehose/EventReceiverFirehoseFactory.java

```diff
@@ -280,52 +294,36 @@ private Thread createDelayedCloseExecutor()
             // The closed = true is visible after close() because there is a happens-before edge between
             // delayedCloseExecutor.interrupt() call in close() and catching InterruptedException below in this loop.
             while (!closed) {
-              Long closeTimeNs = null;
-              Boolean dueToShutdownRequest = null;
+              // Reading idleCloseTimeNs and requestedShutdownTimeNs to locals for absolute confidence that there
+              // couldn't be NPE in comparisons with nanoTime() below.
```

Comment:
I don't see any place where _idleCloseTimeNs/requestedShutdownTimeNs_ can be set to null after it was initialized. So if you check it for `!= null` it stays that way always, meaning I don't understand why additional locals are used.